File: 2026-09/src/day_13/leet_338/Solution.java
package day_13.leet_338;

class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[1 + n];
        for (int i = 1; i <= n; i++) {
            int x = i;
            int cnt = 0;
            while (x > 0) {
                if ((x & 1) == 1) cnt++;
                x >>= 1;
            }
            ans[i] = cnt;
        }

        return ans;
    }
}

File: 2026-09/src/day_13/leet_338/Solution.py
class Solution:
    def countBits(self, n: int) -> list[int]:
        ans = [0] * (1 + n)
        for i in range(1, n + 1):
            x = i
            cnt = 0
            while x:
                if x & 1:
                    cnt += 1
                x >>= 1
            ans[i] = cnt

        return ans
